mypy/fastparse.py

```diff
@@ -324,23 +329,26 @@ def visit_Module(self, mod: ast3.Module) -> MypyFile:
     #             stmt* body, expr* decorator_list, expr? returns, string? type_comment)
     # arguments = (arg* args, arg? vararg, arg* kwonlyargs, expr* kw_defaults,
     #              arg? kwarg, expr* defaults)
-    @with_line
     def visit_FunctionDef(self, n: ast3.FunctionDef) -> Union[FuncDef, Decorator]:
-        return self.do_func_def(n)
+        f = self.do_func_def(n)
+        f.set_line(n.lineno, n.col_offset)  # Overrides set_line -- can't use self.set_line
+        return f
 
     # AsyncFunctionDef(identifier name, arguments args,
     #                  stmt* body, expr* decorator_list, expr? returns, string? type_comment)
-    @with_line
     def visit_AsyncFunctionDef(self, n: ast3.AsyncFunctionDef) -> Union[FuncDef, Decorator]:
-        return self.do_func_def(n, is_coroutine=True)
+        f = self.do_func_def(n, is_coroutine=True)
+        f.set_line(n.lineno, n.col_offset)  # Overrides set_line -- can't use self.set_line
+        return f
 
     def do_func_def(self, n: Union[ast3.FunctionDef, ast3.AsyncFunctionDef],
                     is_coroutine: bool = False) -> Union[FuncDef, Decorator]:
         """Helper shared between visit_FunctionDef and visit_AsyncFunctionDef."""
         no_type_check = bool(n.decorator_list and
                              any(is_no_type_check_decorator(d) for d in n.decorator_list))
 
-        args = self.transform_args(n.args, n.lineno, no_type_check=no_type_check)
+        lineno = n.lineno
```

Comment:
Why not do the same for `n.col_offset`? (Probably a bit below when it is actually used.)

Reply:
It's only used when generating errors, so the code paths will be rare.